api/src/files/schemas/study-doc.schema.ts

```diff
@@ -0,0 +1,30 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document, Schema as MongooseSchema } from 'mongoose';
+import { CourseSubject } from './course-subject.schema';
+import { FileUpload } from './file-upload.schema';
+
+@Schema({ timestamps: true })
+export class StudyDoc extends Document {
+  @Prop({ required: true, type: Number, ref: 'FileUpload' })
+  file: FileUpload;
+
+  @Prop()
+  name?: string;
+
+  @Prop()
+  year?: number;
+
+  @Prop({ type: MongooseSchema.Types.ObjectId, ref: 'CourseSubject' })
+  subject?: CourseSubject;
+
+  @Prop()
```

Comment:
```suggestion
  @Prop({ type: [String], required: false })
```
les metadatas et reflection TS ne permettent pas de trouver le type d'un type "complexe" (genre un array), donc il faut lui donner manuellement.